src/test/scala/edu/ie3/mobsim/model/ChargingBehaviorSpec.scala

```diff
@@ -0,0 +1,372 @@
+          // EV with destination supermarket and home charging possible has
+          // SoC = lower thresholds -> wants to charge
+          Quantities.getQuantity(30, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.SHOPPING,
+          supermarketPoi,
+          true,
+          parkingTimeStart.plusMinutes(15),
+          true
+        ),
+        (
+          // EV with destination supermarket and home charging possible has
+          // SoC = upper thresholds -> does not want to charge
+          Quantities.getQuantity(50, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.SHOPPING,
+          supermarketPoi,
+          true,
+          parkingTimeStart.plusMinutes(15),
+          false
+        ),
+        (
+          // EV with destination home and home charging not possible has
+          // SoC = lower threshold -> wants to charge
+          Quantities.getQuantity(40, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.HOME,
+          poiHome,
+          false,
+          parkingTimeStart.plusMinutes(15),
+          true
+        ),
+        (
+          // EV with destination home and home charging not possible has
+          // SoC = upper threshold -> does not want to charge
+          Quantities.getQuantity(85, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.HOME,
+          poiHome,
+          false,
+          parkingTimeStart.plusMinutes(15),
+          false
+        ),
+        (
+          // EV with destination work and home charging not possible has
+          // SoC = lower threshold -> wants to charge
+          Quantities.getQuantity(40, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.WORK,
+          workPoi,
+          false,
+          parkingTimeStart.plusMinutes(15),
+          true
+        ),
+        (
+          // EV with destination work and home charging not possible has
+          // SoC = upper threshold -> does not want to charge
+          Quantities.getQuantity(85, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.WORK,
+          workPoi,
+          false,
+          parkingTimeStart.plusMinutes(15),
+          false
+        ),
+        (
+          // EV with destination supermarket and home charging not possible has
+          // SoC = lower threshold -> wants to charge
+          Quantities.getQuantity(30, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.SHOPPING,
+          supermarketPoi,
+          false,
+          parkingTimeStart.plusMinutes(15),
+          true
+        ),
+        (
+          // EV with destination supermarket and home charging not possible has
+          // SoC = upper threshold -> does not want to charge
+          Quantities.getQuantity(75, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.SHOPPING,
+          supermarketPoi,
+          false,
+          parkingTimeStart.plusMinutes(15),
+          false
+        )
+      )
+
+      forAll(cases) {
+        (
+            SoC,
```

Comment:
Please start variables with lower case letters :) So here, I'd just name it `soc`